core/src/main/scala/kafka/server/RaftReplicaChangeDelegate.scala

```diff
@@ -72,7 +72,8 @@ class RaftReplicaChangeDelegate(helper: RaftReplicaChangeDelegateHelper) {
   def makeLeaders(prevPartitionsAlreadyExisting: Set[MetadataPartition],
                   partitionStates: Map[Partition, MetadataPartition],
                   highWatermarkCheckpoints: OffsetCheckpoints,
-                  metadataOffset: Option[Long]): Set[Partition] = {
+                  metadataOffset: Option[Long],
+                  topicIds: String => Option[Uuid]): Set[Partition] = {
```

Comment:
I guess we could use a strong type here since topicIds are required for KIP-500, but maybe not worth it since we are delegating to `Partition` in the end.